### PR TITLE
Add cypress obc-test migration

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@bipuladh , @vbnrh Please review

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: cypress/cypress-base/support/i18n.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+declare global {
+  namespace Cypress {
+    interface Chainable<Subject> {
+      isPseudoLocalized(): Chainable<Element>;
+      testI18n(selectors?: string[], testIDs?: string[]): Chainable<Element>;
+    }
+  }
+}
+
+Cypress.Commands.add('testI18n', (selectors: string[] = [], testIDs: string[] = []) => {
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply cypress/cypress-base/support/i18n.ts:13]: We are not using and i have removed this

File: cypress/cypress-base/support/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          if (!i18nNotTranslatedAttr) {
+            cy.wrap($el).isPseudoLocalized();
+          }
+        }),
+      );
+    });
+  });
+});
+
+Cypress.Commands.add(
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply cypress/cypress-base/support/i18n.ts:45]: We are not using and i have removed this

File: cypress/cypress-base/support/selectors.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import Loggable = Cypress.Loggable;
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply cypress/cypress-base/support/selectors.ts:2]: Removed this

File: cypress/cypress-base/support/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Cypress Custom Commands
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply cypress/cypress-base/support/README.md:1]: Yes not required, i have removed this

File: cypress/cypress-base/view/details-page.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+export const detailsPage = {
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply cypress/cypress-base/view/details-page.ts:1]: Sure, Done

File: cypress/views/obcPage.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import { projectNameSpace } from '../support/pages/app';
```
[the remark this answers is not in the document]
Comment (TimothyAsirJeyasing) [reply cypress/views/obcPage.ts:1]: Sure, Its done

**Comment by TimothyAsirJeyasing** [comment]
@bipuladh Please review

File: cypress/mocks/storageclass.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+export const testEbsSC = {
+  apiVersion: 'storage.k8s.io/v1',
+  kind: 'StorageClass',
+  metadata: { name: 'test-ebs-sc' },
+  provisioner: 'kubernetes.io/aws-ebs',
+  parameters: { type: 'io1' },
+  reclaimPolicy: 'Retain',
+};
```
Comment (GowthamShanmugam) [inline cypress/mocks/storageclass.ts:16]: We dont need these two mockups, they can be removed or replaced with RBD and CephFs instead.

File: cypress/support/pages/app.ts [hunk 8]
```diff
@@ -0,0 +1,288 @@
+//#import { detailsPage } from '../../views/details-page';
+import { detailsPage } from '../../cypress-base/support/details-page';
```
Comment (GowthamShanmugam) [inline cypress/support/pages/app.ts:2]: This is for developer perspective, we dont need this

File: cypress/support/pageob.ts [hunk 9]
```diff
@@ -0,0 +1,387 @@
+export const devNavigationMenuPO = {
+  pipelines: '[data-test-id="pipeline-header"]',
+  add: '[data-test-id="+Add-header"]',
```
Comment (GowthamShanmugam) [inline cypress/support/pageob.ts:3]: We dont need this also, this is also dev perspective

File: cypress/tests/obc-test.spec.ts [hunk 10]
```diff
@@ -0,0 +1,137 @@
+import { expect } from 'chai';
+import { testName } from '../utils/consts';
+//import { detailsPage } from '../views/details-page';
```
Comment (GowthamShanmugam) [inline cypress/tests/obc-test.spec.ts:3]:
```suggestion

```

File: cypress/tests/obc-test.spec.ts [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+import { detailsPage } from '../cypress-base/support/details-page';
+import { listPage } from '../cypress-base/support/list-page';
+//import { listPage } from '../views/list-page';
```
Comment (GowthamShanmugam) [inline cypress/tests/obc-test.spec.ts:6]:
```suggestion

```

File: cypress/tests/obc-test.spec.ts [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+    cy.visit(obcUrl);
+    obcHandler.deleteBucketClaim();
+    cy.logout();
+  });
+
+  it('Test if Object Bucket Claim details page is rendered correctly', () => {
+    cy.log('Test if OBC is bound');
+    cy.byTestID('resource-status').contains(BOUND, { timeout: MINUTE });
+
+    cy.log('Test if owner and creation date are shown correctly');
```
Comment (GowthamShanmugam) [inline cypress/tests/obc-test.spec.ts:49]:
```suggestion
    
```

File: cypress/mocks/storageclass.ts (hunk 7 above)
Comment (TimothyAsirJeyasing) [reply cypress/mocks/storageclass.ts]: Done

File: cypress/support/pages/app.ts (hunk 8 above)
Comment (TimothyAsirJeyasing) [reply cypress/support/pages/app.ts]: Done

File: cypress/support/pageob.ts (hunk 9 above)
Comment (TimothyAsirJeyasing) [reply cypress/support/pageob.ts]: Done

File: cypress/tests/obc-test.spec.ts (hunk 10 above)
Comment (TimothyAsirJeyasing) [reply cypress/tests/obc-test.spec.ts]: Done

File: cypress/tests/obc-test.spec.ts (hunk 12 above)
Comment (TimothyAsirJeyasing) [reply cypress/tests/obc-test.spec.ts]: Done

**Comment by TimothyAsirJeyasing** [comment]
@bipuladh , @GowthamShanmugam , @SanjalKatiyar Please review

**Comment by TimothyAsirJeyasing** [comment]
Screenshot from 2022-05-17 13-37-53

**Pull request merged**